src/scenes/Partners/Detail/Tabs/Projects/PartnerDetailProjects.tsx

```diff
@@ -45,14 +48,35 @@ export const PartnerDetailProjects = () => {
     [props.slotProps]
   );
 
+  const getStatusColumnIndex = (field: string, defaultIndex?: number) => {
+    const index = ProjectColumns.findIndex((column) => column.field === field);
+    return index === -1 ? defaultIndex : index + 1;
+  };
+
+  const ProjectPartnerColumns: GridColDef[] = [
+    ...ProjectColumns.slice(0, getStatusColumnIndex('status', 5)),
+    {
+      field: 'partnerships.types',
+      ...multiSelectColumn(PartnerTypeList, PartnerTypeLabels),
+      headerName: 'Partnership Type',
+      width: 160,
+      sortable: false,
+      serverFilter: ({ value }) => ({
+        partnerships: { types: typeof value === 'string' ? [value] : value },
+      }),
+      valueGetter: (_, { partnership }) => partnership.types.value,
+    },
+    ...ProjectColumns.slice(getStatusColumnIndex('status', 0)),
+  ];
+
```

Comment:
@CarsonF  Hope you can help us here. We tried for a long time to use the .splice method here but the type errors we encountered with it were beyond us.   Andrew gave it a go as well but wasn't able to figure it out. 

Here is what we attempted to do: 

```tsx
import {
  PartnerProjectDataGridRowFragment as PartnerProject
} from './PartnerProjects.graphql';
```

```tsx
const partnershipTypeColumn = {
    field: 'partnerships.types',
    ...multiSelectColumn(PartnerTypeList, PartnerTypeLabels),
    headerName: 'Partnership Type',
    width: 160,
    sortable: false,
    serverFilter: ({ value }) => ({ name: value }),
    valueGetter: (_, { partnership }) => partnership.types.value,
  } satisfies GridColDef<PartnerProject>;

  const statusColumnIndex =
    ProjectColumns.findIndex((column) => column.field === 'status') + 1;

  const ProjectPartnerColumns = [
    ...splice(ProjectColumns, statusColumnIndex, 0, partnershipTypeColumn),
  ];
  ```

Reply:
Actually there's a builtin `.toSpliced()` method now, but our `splice()` should do the same thing.
```ts
['a', 'b', 'c', 'd', 'e'].toSpliced(2, 0, 'z')
=> ['a', 'b', 'z', 'c', 'd', 'e']
```